platform/pulp/download/http.py

```diff
@@ -0,0 +1,274 @@
+            response.status_code = status
+        return response
+
+    def close(self):
+        """
+        Close the handler.
+        """
+        pass
+
+
+class HttpDownload(Download):
+    """
+    An HTTP download.
+
+    Attributes:
+        proxy_url (str): An optional proxy URL.
+        headers (dict): The optional HTTP headers.
+
+    Examples:
+        >>>
+        >>> from pulp.download import HttpDownload
+        >>>
+        >>> url = ...
+        >>> path = ...
+        >>> download = HttpDownload(url, path)
+        >>> download()
+        >>> # Go read the downloaded file \o/
+        >>>
+
+    Notes:
+        The 'session' may be shared through the context.session.
+    """
+
+    @staticmethod
+    def create_session():
+        """
+        Create a session.
+
+        Returns:
+            Session: The created session.
+
+        """
+        session = Session()
+        adapter = HTTPAdapter(max_retries=Retry())
+        session.mount('file://', LocalAdapter())
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+        return session
+
+    def __init__(self, url, path=None, method='GET'):
+        """
+        Args:
+            url (str): A file download URL.
+            path (str): The storage path for the downloaded file.
+        """
+        super(HttpDownload, self).__init__(url, path)
+        self.method = self._find_method(method)
+        self.proxy_url = None
+        self.headers = None
+
+    @property
+    def status(self):
+        """
+        The status code in the reply.
+
+        Returns:
+            (int): The reply status.
+
+        """
+        try:
+            return self.reply.status_code
+        except AttributeError:
+            return 0
+
+    @property
+    def session(self):
+        """
+        The `downloads` session.
+
+        Returns:
+            Session: The shared or newly created session.
+
+        Notes:
+            The session can be shared between download but this needs to be
+            facilitated by a 3rd object by setting the `shared` dictionary
+            to be the same.
+        """
+        with self.context as context:
+            try:
+                return context.session
+            except AttributeError:
+                session = self.create_session()
+                context.session = session
+                return session
+
+    def _find_method(self, name):
+        """
+        Find an http method by name.
+
+        Args:
+            name: The method name.  see: METHODS.
+
+        Returns:
+            instancemethod: When matched.
+
+        Raises:
+            ValueError: When not matched.
+        """
+        methods = {
+            'GET': self.get,
+            'HEAD': self.head,
+        }
+        try:
+            return methods[name.upper()]
+        except KeyError:
+            _list = '|'.join(sorted(methods.keys()))
+            msg = _('method must be: ({list})'.format(list=_list))
```

Comment:
This resolving the HTTP method (example: GET) passed into the constructor as `method` to an implementation.  I think the `ValueError` raised will be clearer this way.
Example:
```
d = HttpDownload(url, method='BAD')
```
would raise a `ValueError` something like:
```
ValueError: "method must be (GET, HEAD)"
```